Doc/whatsnew/3.7.rst

```diff
@@ -446,35 +476,304 @@ Other Language Changes
+
+contextvars
+-----------
+
+The new :mod:`contextvars` module and a set of new C APIs introduce
+support for *context variables*.  Context variables are conceptually
+similar to thread-local variables.  Unlike TLS, context variables
+support asynchronous code correctly.
+
+The :mod:`asyncio` and :mod:`decimal` modules have been updated to use
+and support context variables out of the box.  Particularly the active
+decimal context is now stored in a context variable, which allows
+decimal operations to work with the correct context in asynchronous code.
+
+.. seealso::
+
+    :pep:`567` -- Context Variables
+       PEP written and implemented by Yury Selivanov
+
+
+.. _whatsnew37-pep557:
+
+dataclasses
+-----------
+
+The new :func:`~dataclasses.dataclass` decorator provides a way to declare
+*data classes*.  A data class describes its attributes using class variable
+annotations.  Its constructor and other magic methods, such as
+:meth:`~object.__repr__`, :meth:`~object.__eq__`, and
+:meth:`~object.__hash__` are generated automatically.
+
+Example::
+
+    @dataclass
+    class Point:
+        x: float
+        y: float
+        z: float = 0.0
+
+    p = Point(1.5, 2.5)
+    print(p)   # produces "Point(x=1.5, y=2.5, z=0.0)"
+
+.. seealso::
+
+    :pep:`557` -- Data Classes
+       PEP written and implemented by Eric V. Smith
+
+
+.. _whatsnew37_importlib_resources:
+
 importlib.resources
 -------------------
 
-This module provides several new APIs and one new ABC for access to, opening,
-and reading *resources* inside packages.  Resources are roughly akin to files
-inside of packages, but they needn't be actual files on the physical file
-system.  Module loaders can provide a :meth:`get_resource_reader()` function
-which returns a :class:`importlib.abc.ResourceReader` instance to support this
+The new :mod:`importlib.resources` module provides several new APIs and one
+new ABC for access to, opening, and reading *resources* inside packages.
+Resources are roughly similar to files inside packages, but they needn't
+be actual files on the physical file system.  Module loaders can provide a
+:meth:`get_resource_reader()` function which returns
+a :class:`importlib.abc.ResourceReader` instance to support this
 new API.  Built-in file path loaders and zip file loaders both support this.
-(see the PyPI package
-`importlib_resources <http://importlib-resources.readthedocs.io/en/latest/>`_
-as a compatible back port for older Python versions).
+
+Contributed by Barry Warsaw and Brett Cannon in :issue:`32248`.
+
+.. seealso::
+
+    `importlib_resources <http://importlib-resources.readthedocs.io/en/latest/>`_
+    -- a PyPI backport for earlier Python versions.
 
 
 Improved Modules
 ================
 
 
+abc
+---
+
+Improved conistency of exceptions raised by :func:`issubclass` when called
+with a non-class and an abstract base class as the first and second
+arguments, respectively.  (Contributed by Josh Bronson in :issue:`33018`.)
```

Comment:
I'd also add a ref to optimizations section's paragraph that covers `ABCMeta` being rewritten in C.

Reply:
Note the "conistency"